x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/import/action_connectors/import_rule_action_connectors.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Readable } from 'stream';
+
+import type { SavedObjectsImportResponse } from '@kbn/core-saved-objects-common';
+import type { SavedObject } from '@kbn/core-saved-objects-server';
+
+import type { WarningSchema } from '../../../../../../../common/detection_engine/schemas/response';
+import {
+  checkIfActionsHaveMissingConnectors,
+  filterExistingActionConnectors,
+  getActionConnectorRules,
+  handleActionsHaveNoConnectors,
+  mapSOErrorToRuleError,
+  returnErroredImportResult,
+} from './utils';
+import type { ImportRuleActionConnectorsParams, ImportRuleActionConnectorsResult } from './types';
+
+export const importRuleActionConnectors = async ({
+  actionConnectors,
+  actionsClient,
+  actionsImporter,
+  rules,
+  overwrite,
+}: ImportRuleActionConnectorsParams): Promise<ImportRuleActionConnectorsResult> => {
+  try {
+    const actionConnectorRules = getActionConnectorRules(rules);
+    const actionsIds: string[] = Object.keys(actionConnectorRules);
+
+    if (!actionsIds.length)
+      return {
+        success: true,
+        errors: [],
+        successCount: 0,
+        warnings: [],
+      };
+
+    if (overwrite && !actionConnectors.length)
+      return handleActionsHaveNoConnectors(actionsIds, actionConnectorRules);
+
+    let actionConnectorsToImport: SavedObject[] = actionConnectors;
+
+    if (!overwrite) {
+      const newIdsToAdd = await filterExistingActionConnectors(actionsClient, actionsIds);
+
+      const foundMissingConnectors = checkIfActionsHaveMissingConnectors(
+        actionConnectors,
+        newIdsToAdd,
+        actionConnectorRules
+      );
+      if (foundMissingConnectors) return foundMissingConnectors;
+      // filter out existing connectors
+      actionConnectorsToImport = actionConnectors.filter(({ id }) => newIdsToAdd.includes(id));
+    }
+    if (!actionConnectorsToImport.length)
+      return {
+        success: true,
+        errors: [],
+        successCount: 0,
+        warnings: [],
+      };
+
+    const readStream = Readable.from(actionConnectorsToImport);
+    const { success, successCount, successResults, warnings, errors }: SavedObjectsImportResponse =
+      await actionsImporter.import({
+        readStream,
+        overwrite,
+        createNewCopies: false,
+      });
```

Comment:
I'm guessing this might be a core bug since it's coming straight from the `actionsImporter`, but when importing an `ndjson` with two of the _same type_ of connector requiring sensitive information, it only says there's one connector that needs updated.

E.g. with these two `.slack` connectors in my `ndsjon` export:
<p align="center">
  <img width="125" src="https://user-images.githubusercontent.com/2946766/215906533-0f1db3d0-7d2d-41e8-9ecc-f28de804c533.png" />
</p>


When I import I see it say only `1 connector` requires additional information:


<p align="center">
  <img width="300" src="https://user-images.githubusercontent.com/2946766/215906603-41a84b77-1e2c-4bd3-bdcd-e9a958893388.png" />
</p>
But when clicking through connectors you see there's two:


<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/215906680-e3b5b958-5651-467d-8f02-d784c6fe9604.png" />
</p>

Reply:
That's really a good catch thank you! I found the reason usually I was reusing the same connector type but in this test case slack has been used to create 2 different connectors.


The reason is I discard the message from the warning object returned by the `Core`, and used a new message as per the UX as I was expecting that I will get warning objects equivalent to the number of failing connectors because it is an array however I got 1 object with this message `2 connectors have sensitive information that require updates`

Reply:
The same translation logic is handled in `x-pack/plugins/actions/server/saved_objects/get_import_warnings.ts` so I believe I will use the message from one object and indeed is one object in the array :D

Reply:
Please validate https://github.com/elastic/kibana/pull/148703/commits/31f596325ff0e726a684b5897a75830833a6dc08#diff-5de0ca9e980410c1737603f031129590e8f5eec2c20db0d490cd3ce2df6eaaf6R26